util/stopwaiter/stopwaiter_test.go

```diff
@@ -0,0 +1,89 @@
+// Copyright 2021-2022, Offchain Labs, Inc.
+// For license information, see https://github.com/nitro/blob/master/LICENSE
+
+package stopwaiter
+
+import (
+	"context"
+	"fmt"
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/ethereum/go-ethereum/log"
+	"github.com/offchainlabs/nitro/util/testhelpers"
+)
+
+const testStopDelayWarningTimeout = 350 * time.Millisecond
+
+type TestStruct struct{}
+
+func TestStopWaiterStopAndWaitTimeoutShouldWarn(t *testing.T) {
+	logHandler := initTestLog(t, log.LvlTrace)
+	sw := StopWaiter{}
+	sw.Start(context.Background(), &TestStruct{})
+	sw.LaunchThread(func(ctx context.Context) {
+		for {
+			select {
+			case <-ctx.Done():
+				return
+			default:
+				time.Sleep(testStopDelayWarningTimeout + 150*time.Millisecond)
+			}
+		}
+	})
+	time.Sleep(50 * time.Millisecond)
+	sw.stopAndWaitImpl(testStopDelayWarningTimeout)
+	if !logHandler.WasLogged(fmt.Sprintf("stopwaiter.TestStruct taking more than %s to stop", testStopDelayWarningTimeout.String())) {
+		testhelpers.FailImpl(t, "Failed to log about waiting long on StopAndWait")
+	}
+}
+
+func TestStopWaiterStopAndWaitTimeoutShouldNotWarn(t *testing.T) {
+	logHandler := initTestLog(t, log.LvlTrace)
+	sw := StopWaiter{}
+	sw.Start(context.Background(), &TestStruct{})
+	sw.LaunchThread(func(ctx context.Context) {
+		for {
+			select {
+			case <-ctx.Done():
+				return
+			default:
+				time.Sleep(200 * time.Millisecond)
+			}
+		}
+	})
+	time.Sleep(50 * time.Millisecond)
+	sw.StopAndWait()
+	if logHandler.WasLogged(fmt.Sprintf("stopwaiter.TestStruct taking more than %s to stop", stopDelayWarningTimeout.String())) {
+		testhelpers.FailImpl(t, "Incorrectly logged about waiting long on StopAndWait")
+	}
+}
+
+type LogHandler struct {
```

Comment:
LogHandler is great and will be very useful.
I suggest putting it in util/testhelpers